packages/salesforcedx-vscode-apex/test/vscode-integration/requirements.test.ts

```diff
@@ -8,29 +8,51 @@
 // tslint:disable:no-unused-expression
 
 import { expect } from 'chai';
-import * as path from 'path';
-import * as shell from 'shelljs';
-import { workspace } from 'vscode';
-import { JAVA_HOME_KEY, JAVA_MEMORY_KEY } from '../../src/requirements';
+import { createSandbox, SinonSandbox, SinonStub } from 'sinon';
+import * as vscode from 'vscode';
+import { JAVA_HOME_KEY, resolveRequirements } from '../../src/requirements';
+import pathExists = require('path-exists');
+import * as cp from 'child_process';
+
+const runtimePath = '~/java_home/real/jdk/openjdk1.8.0.302_8.56.0.22_x64';
 
 describe('Java Requirements Test', () => {
-  it('The jar should be signed', () => {
-    shell.config.execPath = process.execPath;
-    const apexJarPath = path.join(__dirname, '..', '..', 'apex-jorje-lsp.jar');
-    expect(
-      shell
-        .exec(`jarsigner -verify ${apexJarPath}`)
-        .stdout.includes('jar verified')
-    ).to.be.true;
+  let sandbox: SinonSandbox;
+  let settingStub: SinonStub;
+  let pathExistsStub: SinonStub;
+  let execFileStub: SinonStub;
+
+  beforeEach(() => {
```

Comment:
Ah ha, that is a good callout! I actually wanted to move this to a unit testing directory, but the Apex package doesn't have one yet, and it was proving a beast to set up last night. I can give that another try though, because ideally we would have a unit test section (which would also allow this test to run a LOT faster).

Reply:
Thanks for the clarification. If we have issues moving it out, we can create a TODO item and come back to it when possible.

Reply:
I think I'll go ahead and do that here; if we have additional unit tests we want to add, it would be good to combine it with some other changes.